api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -121,14 +120,14 @@ type IngressNodeFirewallSpec struct {
 	// empty list indicates no ingress firewall i.e allow all incoming traffic.
 	// +kubebuilder:validation:Optional
 	// +optional
-	Ingress []IngressNodeFirewallRules `json:"ingress"`
+	Ingress []IngressNodeFirewallRules `json:"ingress,omitempty"`
 
 	// A list of interfaces where the ingress firewall policy will be applied on.
 	// empty list indicates the firewall policy applied on all interfaces
 	// +kubebuilder:validation:Optional
 	// +optional
 	// +nullable
-	Interfaces *[]string `json:"interfaces"`
+	Interfaces *[]string `json:"interfaces,omitempty"`
```

Comment:
A slice is already a pointer. What's the benefit if having `*[]`?

Reply:
it allow nullable the fields need to be pointer and that was discussed in the enhancement nil vs empty list @danwinship suggested to use `nullable` field

Reply:
you can set []string = nil, []string is nullable

Reply:
Ok gotcha, it's for server side apply, makes sense then, sorry for the noise!
https://go.dev/play/p/PWfHvr-Ut8W